test/OrchardCore.Tests/OrchardCore.Users/UserValidatorTests.cs

```diff
@@ -0,0 +1,150 @@
+            Assert.Equal(describer.InvalidEmail(user.Email).Code, result.Errors.First().Code);
+        }
+
+        [Fact]
+        public async Task ShouldRequireValidEmail()
+        {
+            // Arrange
+            var describer = new IdentityErrorDescriber();
+            var userManager = UsersMockHelper.MockUserManager<IUser>();
+            var user = new User { UserId = Guid.NewGuid().ToString("n"), UserName = "Foo", Email = "foo" };
+            userManager.Setup(m => m.GetUserIdAsync(user)).ReturnsAsync(user.UserId);
+            userManager.Setup(m => m.GetUserNameAsync(user)).ReturnsAsync(user.UserName);
+            userManager.Setup(m => m.GetEmailAsync(user)).ReturnsAsync(user.Email);
+
+            // Act
+            var validator = userManager.Object.UserValidators.FirstOrDefault();
+            var result = await validator.ValidateAsync(userManager.Object, user);
+
+            // Test
+            Assert.False(result.Succeeded);
+            Assert.Equal(describer.InvalidEmail(user.Email).Code, result.Errors.First().Code);
+        }
+
+        [Fact]
+        public async Task ShouldRequireUniqueEmail()
+        {
+            // Arrange
+            var describer = new IdentityErrorDescriber();
+            var userManager = UsersMockHelper.MockUserManager<IUser>();
+            var existingUser = new User { UserId = Guid.NewGuid().ToString("n"), UserName = "Foo", Email = "foo@foo.com" };
+            userManager.Setup(m => m.GetUserIdAsync(existingUser)).ReturnsAsync(existingUser.UserId);
+            userManager.Setup(m => m.GetUserNameAsync(existingUser)).ReturnsAsync(existingUser.UserName);
+            userManager.Setup(m => m.GetEmailAsync(existingUser)).ReturnsAsync(existingUser.Email);
+            userManager.Setup(m => m.FindByEmailAsync(existingUser.Email)).ReturnsAsync(existingUser);
+            userManager.Setup(m => m.FindByNameAsync(existingUser.UserName)).ReturnsAsync(existingUser);
+
+            var user = new User { UserId = Guid.NewGuid().ToString("n"), UserName = "Bar", Email = "foo@foo.com" };
+            userManager.Setup(m => m.GetUserIdAsync(user)).ReturnsAsync(user.UserId);
+            userManager.Setup(m => m.GetUserNameAsync(user)).ReturnsAsync(user.UserName);
+            userManager.Setup(m => m.GetEmailAsync(user)).ReturnsAsync(user.Email);
+
+            // Act
+            var validator = userManager.Object.UserValidators.FirstOrDefault();
+            var result = await validator.ValidateAsync(userManager.Object, user);
+
+            // Test
+            Assert.False(result.Succeeded);
+            Assert.Equal(describer.DuplicateEmail(user.Email).Code, result.Errors.First().Code);
+        }
+
+        [Fact]
+        public async Task ShouldRequireUserNameIsNotAnEmailAddress()
+        {
+            // Arrange
+            var describer = new IdentityErrorDescriber();
+            var userManager = UsersMockHelper.MockUserManager<IUser>();
```

Comment:
I kept the tests, although identity also has these tests.

Mostly just to indicate that we have specified a requirement.

They could go if not wanted